contract_forecast/models/contract_line.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2019 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from dateutil.relativedelta import relativedelta
+from odoo import api, fields, models
+from odoo.addons.queue_job.job import job
+
+QUEUE_CHANNEL = "root.CONTRACT_FORECAST"
+
+
+class AccountAnalyticInvoiceLine(models.Model):
+
+    _inherit = "account.analytic.invoice.line"
+
+    forecast_period_ids = fields.One2many(
+        comodel_name="contract.line.forecast.period",
+        inverse_name="contract_line_id",
+        string="Forecast Periods",
+        required=False,
```

Comment:
@sbejaoui @aheficent no need `required=False`, By default `False`

Reply:
@nikul-serpentcs Code is never too explicit, let's keep it as it :slightly_smiling_face: